### DISCUSSION TITLE
how to check bram after synthesis

### DISCUSSION THREAD

**Discussion by nthuyouwei** [bol-edu/HLS-SOC-Discussions#96] [Q&A]
我不確定在做完synthesis 後，如何確認有沒有用到bram。
第一個是在synthesis report 中的part resource 下的BRAMs
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145022311/58bba0fa-ae81-4d47-abe4-ec6693ad5081)
第二是Utizilation 中 memory 的部分。
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145022311/53f9638a-c4a7-4e0c-b1b4-468265a1f684)
最後我認為我內部應該沒有用到brams，但不知道為甚麼第一個有顯示，還是那是指外部的。

**Comment by LingChiYang** [comment]
同學好

第一張圖顯示的是可使用的bram，也就是pynq上bram的總數(但不一定有使用)。第二張圖才是真正使用的。

**Reply by nthuyouwei** [reply]
那想請問那第一張中的加法器跟乘法器，是指我使用的嗎還是一樣是pynq的

**Reply by LingChiYang** [reply]
使用的。你也可以從第二張圖的report中找到相關訊息

**Comment by boledulab** [comment]
另更真實點是去看產生出來的 gate-netlist 中有沒有 BRAM.
